android/app/src/main/java/com/microsoft/codepush/react/CodePushDialog.java

```diff
@@ -16,9 +17,43 @@ public CodePushDialog(ReactApplicationContext reactContext) {
     }
 
     @ReactMethod
-    public void showDialog(String title, String message, String button1Text, String button2Text,
-                      final Callback successCallback, Callback errorCallback) {
-        AlertDialog.Builder builder = new AlertDialog.Builder(getCurrentActivity());
+    public void showDialog(final String title, final String message, final String button1Text,
+                           final String button2Text, final Callback successCallback, Callback errorCallback) {
+        Activity currentActivity = getCurrentActivity();
+        if (currentActivity == null) {
+            // If getCurrentActivity is null, it could be because the app is backgrounded,
+            // so we show the dialog when the app resumes)
+            getReactApplicationContext().addLifecycleEventListener(new LifecycleEventListener() {
+                private boolean shown = false;
+
+                @Override
+                public void onHostResume() {
+                    Activity currentActivity = getCurrentActivity();
+                    if (!shown && currentActivity != null) {
+                        shown = true;
```

Comment:
I don't think we need the `shown` variable here - this listener will be removed as soon as it's shown